src/Widget/Widget.php

```diff
@@ -91,98 +92,44 @@ public static function end()
-     * The widget rendering result is returned by this method.
-     *
-     * @param array $config name-value pairs that will be used to initialize the object properties
-     *
-     * @return string the rendering result of the widget.
-     * @throws \Exception
+     * Creates a widget instance.
      *
+     * @return Widget $widget.
      */
-    public static function widget($config = []): string
+    public static function widget(): Widget
     {
-        ob_start();
-        ob_implicit_flush(false);
-
-        try {
-            /* @var $widget Widget */
-            $config['__class'] = static::class;
-            $widget = Yii::createObject($config);
-            $out = '';
-            if ($widget->beforeRun()) {
-                $result = $widget->run();
-                $out = $widget->afterRun($result);
-            }
-        } catch (\Exception $e) {
-            // close the output buffer opened above if it has not been closed already
-            if (ob_get_level() > 0) {
-                ob_end_clean();
-            }
+        $widget = new static(self::$eventDispatcher, self::$webView);
 
-            throw $e;
-        }
-
-        return ob_get_clean() . $out;
-    }
+        static::$widget = $widget;
 
-    private $id;
-
-    /**
-     * Returns the ID of the widget.
-     *
-     * @param bool $autoGenerate whether to generate an ID if it is not set previously
-     *
-     * @return string ID of the widget.
-     */
-    public function getId($autoGenerate = true): string
-    {
-        if ($autoGenerate && $this->id === null) {
-            $this->id = static::$autoIdPrefix . static::$counter++;
-        }
-
-        return $this->id;
+        return $widget;
     }
 
     /**
-     * Sets the ID of the widget.
+     * Returns the view object that can be used to render views or view files.
      *
-     * @param string $value id of the widget.
+     * The {@see render()} and {@see renderFile()} methods will use this view object to implement the actual view
+     * rendering. If not set, it will default to the "view" application component.
      */
-    public function setId(string $value): void
+    public function getView(): WebView
     {
-        $this->id = $value;
+        return self::$webView;
     }
 
-    private $view;
-
-    /**
-     * Returns the view object that can be used to render views or view files.
-     * The [[render()]] and [[renderFile()]] methods will use
-     * this view object to implement the actual view rendering.
-     * If not set, it will default to the "view" application component.
-     */
-    public function getView(): WebView
+    public function init(): void
     {
-        return $this->view;
     }
 
-    /**
-     * Sets the view object to be used by this widget.
-     *
-     * @param WebView $view the view object that can be used to render views or view files.
-     */
-    public function setView(WebView $view): void
+    public function show(): string
```

Comment:
What is the intent of this method?

Reply:
```
public function run(): string
{
    $out = '';
    $widget = static::$widget;

    if ($widget->beforeRun()) {
        $result = $widget->show();
        $out = $widget->afterRun($result);
    }
    return $out;
}
```

Reply:
I see. We should come up with a better name. "Show" usually stands for revealing something hidden.

Reply:
How about `getContent`?

Reply:
Done.